bundles/org.eclipse.packagedrone.utils.rpm/src/org/eclipse/packagedrone/utils/rpm/parse/RpmInputStream.java

```diff
@@ -100,51 +98,52 @@ protected void ensureInit () throws IOException
     private InputStream setupPayloadStream () throws IOException
     {
         final Object payloadFormatValue = this.payloadHeader.getTag ( RpmTag.PAYLOAD_FORMAT );
-        final Object payloadCodingValue = this.payloadHeader.getTag ( RpmTag.PAYLOAD_CODING );
 
         if ( payloadFormatValue != null && ! ( payloadFormatValue instanceof String ) )
         {
             throw new IOException ( "Payload format must be a single string" );
         }
 
-        if ( payloadFormatValue != null && ! ( payloadCodingValue instanceof String ) )
-        {
-            throw new IOException ( "Payload coding must be a single string" );
-        }
-
         String payloadFormat = (String)payloadFormatValue;
-        String payloadCoding = (String)payloadCodingValue;
 
-        if ( payloadFormat == null || payloadFormat.isEmpty () )
+        if ( payloadFormat == null )
         {
             payloadFormat = "cpio";
         }
 
-        if ( payloadCoding == null || payloadCoding.isEmpty () )
+        if ( !"cpio".equals ( payloadFormat ) )
         {
-            payloadCoding = "gzip";
+            throw new IOException ( String.format ( "Unknown payload format: %s", payloadFormat ) );
         }
 
-        if ( !"cpio".equals ( payloadFormat ) )
+        final Optional<Object> payloadCodingHeader = this.payloadHeader.getOptionalTag ( RpmTag.PAYLOAD_CODING );
+
+        if ( !payloadCodingHeader.isPresent () )
         {
-            throw new IOException ( String.format ( "Unknown payload format: %s", payloadFormat ) );
+            return this.in;
         }
 
-        switch ( payloadCoding )
+        final Object payloadCodingValue = payloadCodingHeader.get ();
+
+        String payloadCoding = (String)payloadCodingValue;
+
+        if ( payloadCodingValue != null && ! ( payloadCodingValue instanceof String ) )
+        {
+            throw new IOException ( "Payload coding must be a single string" );
+        }
+
+        final PayloadCoding coding;
+
+        if ( payloadCoding == null )
         {
-            case "none":
-                return this.in;
-            case "gzip":
-                return new GzipCompressorInputStream ( this.in );
-            case "bzip2":
-                return new BZip2CompressorInputStream ( this.in );
-            case "lzma":
-                return new LZMAInputStream ( this.in );
-            case "xz":
-                return new XZInputStream ( this.in );
-            default:
-                throw new IOException ( String.format ( "Unknown coding: %s", payloadCoding ) );
+            coding = PayloadCoding.GZIP;
```

Comment:
If the payload coding header exists, but the value is `null`, we return `GZIP`. However, I believe that the upstream RPM may allow passthrough, supporting both `GZIP` and `NONE` with a `null` value.